src/Analyzer.JsonRuleEngine.UnitTests/InequalityOperatorTests.cs

```diff
@@ -0,0 +1,160 @@
+        [DataRow(1.3, 2.8, true, true, false, DisplayName = "A float is not greater or equal to another float")]
+        [DataRow(2.8, 2.8, true, true, true, DisplayName = "A float is greater or equal to another float because both are equal")]
+        // <=float
+        [DataRow(1.3, 2.8, false, true, true, DisplayName = "A float is less or equal to another float")]
+        [DataRow(2.8, 1.3, false, true, false, DisplayName = "A float is not less or equal to another float")]
+        [DataRow(1.3, 1.3, false, true, true, DisplayName = "A float is less or equal to another float because both are equal")]
+        public void EvaluateExpression_ValidNumericType_ReturnsExpectedEvaluationResult(object leftValue, object rightValue, bool greater, bool orEquals, bool evaluationResult)
+        {
+            CompareObjects(leftValue, rightValue, greater, orEquals, evaluationResult);
+        }
+
+        [DataTestMethod]
+        // >date
+        [DataRow(637676928000000000, 637500672000000000, true, false, true, DisplayName = "A date is greater than another date")]
+        [DataRow(637500672000000000, 637676928000000000, true, false, false, DisplayName = "A date is not greater than another date")]
+        [DataRow(637500672000000000, 637500672000000000, true, false, false, DisplayName = "A date is not greater than another date because both are equal")]
+        // <date
+        [DataRow(637500672000000000, 637676928000000000, false, false, true, DisplayName = "A date is less than another date")]
+        [DataRow(637676928000000000, 637500672000000000, false, false, false, DisplayName = "A date is not less than another date")]
+        [DataRow(637676928000000000, 637676928000000000, false, false, false, DisplayName = "A date is not less than another date because both are equal")]
+        // >=date
+        [DataRow(637676928000000000, 637500672000000000, true, true, true, DisplayName = "A date is greater or equal to another date")]
+        [DataRow(637500672000000000, 637676928000000000, true, true, false, DisplayName = "A date is not greater or equal to another date")]
+        [DataRow(637676928000000000, 637676928000000000, true, true, true, DisplayName = "A date is greater or equal to another date because both are equal")]
+        // <=date
+        [DataRow(637500672000000000, 637676928000000000, false, true, true, DisplayName = "A date is less or equal to another date")]
+        [DataRow(637676928000000000, 637500672000000000, false, true, false, DisplayName = "A date is not less or equal to another date")]
+        [DataRow(637500672000000000, 637500672000000000, false, true, true, DisplayName = "A date is less or equal to another date because both are equal")]
+        public void EvaluateExpression_ValidDateType_ReturnsExpectedEvaluationResult(long leftTicks, long rightTicks, bool greater, bool orEquals, bool evaluationResult)
```

Comment:
There are functional parsing tests in the project Analyzer.JsonRuleEngine.FunctionalTests in RuleParsingTests.cs.  I think it would be valuable to add some tests parsing at least date values in this operator.

Reply:
While doing this I realized there might be a bigger issue. I didn't realize there are many ways to specify dates in ARM templates, but JToken doesn't seem to parse all of them :( Working on that

Reply:
@yane3628 @JohnathonMohr I'm not sure how to handle dates specified using [these functions](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-date). Should I add support for that in this PR? Do we want to add support for dates comparisons without handling those cases?

I also haven't been able to find a complete documentation on date formates supported by ARM templates, besides 2021-02-28T18:17:16.543Z and 2021-02-28, I assume supporting only those two formats is not enough, right?

Reply:
dateTimeAdd and other functions are listed as another issue https://github.com/Azure/template-analyzer/issues/88.

The docs also mention this format 20190305T175318Z

Reply:
I went through all (😵) the quick-start templates, and the documentation of the resources I found that specify dates, as Johnathon suggested, and came up with the accepted formats in the documentation

I also used [this ](https://docs.microsoft.com/en-us/rest/api/storageservices/formatting-datetime-values) as a guide of how to express it

I know the [date functions](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-date) allow a ton more formats but I suppose that's out of scope for this PR